ReportUtilityBills/measure.rb

```diff
@@ -0,0 +1,785 @@
+    building_id = @model.getBuilding.additionalProperties.getFeatureAsString('building_id').get
+    hpxml = HPXML.new(hpxml_path: hpxml_defaults_path, building_id: building_id)
+
+    # Require full annual simulation
+    if !(hpxml.header.sim_begin_month == 1 && hpxml.header.sim_begin_day == 1 && hpxml.header.sim_end_month == 12 && hpxml.header.sim_end_day == 31)
+      runner.registerError('A full annual simulation is required for calculating utility bills.')
+      return false
+    end
+
+    # Require user-specified utility rate if 'User-Specified'
+    if args[:electricity_bill_type] == 'Detailed' && args[:electricity_utility_rate_type].get == 'User-Specified' && !args[:electricity_utility_rate_user_specified].is_initialized
+      runner.registerError('Must specify a utility rate json path when choosing User-Specified utility rate type.')
+      return false
+    end
+
+    # Require not DSE
+    (hpxml.heating_systems + hpxml.heat_pumps).each do |htg_system|
+      next unless (htg_system.is_a?(HPXML::HeatingSystem) && htg_system.is_heat_pump_backup_system) || htg_system.fraction_heat_load_served > 0
+      next if htg_system.distribution_system_idref.nil?
+      next unless htg_system.distribution_system.distribution_system_type == HPXML::HVACDistributionTypeDSE
+      next if htg_system.distribution_system.annual_heating_dse.nil?
+
+      runner.registerError('DSE is not currently supported when calculating utility bills.')
+      return false
+    end
+    (hpxml.cooling_systems + hpxml.heat_pumps).each do |clg_system|
+      next unless clg_system.fraction_cool_load_served > 0
+      next if clg_system.distribution_system_idref.nil?
+      next unless clg_system.distribution_system.distribution_system_type == HPXML::HVACDistributionTypeDSE
+      next if clg_system.distribution_system.annual_cooling_dse.nil?
+
+      runner.registerError('DSE is not currently supported when calculating utility bills.')
+      return false
+    end
+
+    # Set paths
+    output_dir = File.dirname(@sqlFile.path.to_s)
+    output_path = File.join(output_dir, "results_bills.#{output_format}")
+
+    # Setup outputs
+    fuels, utility_rates, utility_bills = setup_outputs()
+
+    # Get timestamps
+    @timestamps = get_timestamps()
+
+    # Get outputs
+    get_outputs(fuels)
+
+    # Get utility rates
+    get_utility_rates(fuels, utility_rates, args, hpxml.header.state_code, hpxml.pv_systems, runner)
+
+    # Calculate utility bills
+    net_elec = get_utility_bills(fuels, utility_rates, utility_bills, args, hpxml.header.sim_calendar_year)
+
+    # Annual true up
+    annual_true_up(utility_rates, utility_bills, net_elec)
+
+    # Calculate annual bills
+    get_annual_bills(utility_bills)
+
+    # Report results
+    utility_bills.each do |fuel_type, utility_bill|
+      utility_bill_type_str = OpenStudio::toUnderscoreCase("#{fuel_type} $")
+      runner.registerValue(utility_bill_type_str, utility_bill.annual_total.round(2))
```

Comment:
Can you also `registerInfo` these values as we do [here](https://github.com/NREL/OpenStudio-HPXML/blob/d3a3bd73ba2c6a398fd722826b608ce5bbe20ba9/ReportSimulationOutput/measure.rb#L1191)? If you then use the `--debug` flag when calling the CLI, you can see the all registered values in the run.log. (It looks like we need this in the ReportHPXMLOutput measure too.)

Reply:
I manually added the registerInfo call to see what comes out:

```
Info: Registering 1305.57 for electricity_.
Info: Registering 223.92 for natural_gas_.
Info: Registering 0.0 for fuel_oil_.
Info: Registering 0.0 for propane_.
Info: Registering 0.0 for wood_cord_.
Info: Registering 0.0 for wood_pellets_.
Info: Registering 0.0 for coal_.
```

It looks like the "$" gets stripped, maybe use usd instead.

Also, it would be nice to register _every_ value that shows up in the output CSV/JSON, not just fuel totals.